app/views/sweets/_add_sweet_form.erb

```diff
@@ -0,0 +1,12 @@
+<%= form_with model: @vendor_sweet do |f|%>
+  <%= f.hidden_field :vendor_id, value: @vendor.id %>
```

Comment:
`@vendor_sweet` and `@vendor` should be passed as a local instead of using the instance variable.